src/RecoveryChallengeController.js

```diff
@@ -91,7 +92,56 @@ function (Okta, FormController, FormType, FooterSignout) {
       ]
     },
 
+    events: {
+      'click .send-email-link': function (e) {
+        e.preventDefault();
+        var settings = this.model.settings,
+            username = this.options.appState.get('username'),
+            recoveryType = this.options.appState.get('recoveryType');
+
+        this.model.startTransaction(function (authClient) {
+          // The user could have landed here via the Forgot Password/Unlock Account flow
+          switch (recoveryType) {
+            case Enums.RECOVERY_TYPE_PASSWORD:
+              return authClient.forgotPassword({
+                username: settings.transformUsername(username, Enums.FORGOT_PASSWORD),
+                factorType: Enums.RECOVERY_FACTOR_TYPE_EMAIL
+              });
+            case Enums.RECOVERY_TYPE_UNLOCK:
+              return authClient.unlockAccount({
+                username: settings.transformUsername(username, Enums.UNLOCK_ACCOUNT),
+                factorType: Enums.RECOVERY_FACTOR_TYPE_EMAIL
+              });
+            default:
+              return;
+          }
+        });
+      }
+    },
+
     initialize: function () {
+      var recoveryType = this.options.appState.get('recoveryType'),
+          sendEmailLink;
+
+      switch (recoveryType) {
+        case Enums.RECOVERY_TYPE_PASSWORD:
+          sendEmailLink = Okta.loc('password.forgot.sms.notReceived', 'login');
+          break;
+        case Enums.RECOVERY_TYPE_UNLOCK:
+          sendEmailLink = Okta.loc('account.unlock.sms.notReceived', 'login');
+          break;
+        default:
+          break;
+      }
+
+      if (sendEmailLink) {
+        this.add(
+          '<a href="#" class="link send-email-link" data-se="send-email-link">' +
```

Comment:
Let's use a template for this (general rule of thumb to never concatenate html strings with variables)